### DISCUSSION TITLE
Impossible to preserve existing program settings when updating SQLiteStudio

### DISCUSSION THREAD

**Discussion by githubuser181226** [pawelsalawa/sqlitestudio#5141] [Q&A]
### Details
Hi.
Recently I've made an attempt to update SQLiteStudio from 3.4.3 to 3.4.7, but new version immediately crashes when started.
Alas it seems it's the settings3 file created using 3.4.3 that's causing it.
When settings3 is removed the program starts as intended.

Of course this means my custom functions and snippets are all gone, and I have literally hundreds of them.
Surprisingly I don't see any way of importing old settings using updated SQLiteStudio.

Please, tell me how to import all my previous custom settings (functions, snippets etc.), so I don't have to waste hours of time recreating them after every program update.

Thank you.

### Steps to reproduce
1. Create custom function or code snippet using version SQLiteStudio 3.4.3
2. Update SQLiteStudio to 3.4.7 preserving settings3 file from 3.4.3.
3. Run updated SQLiteStudio.

### Operating system
Win7 x64 and Win10 x64

### SQLiteStudio version
3.4.7

**Comment by pawelsalawa** [comment]
It would be nice to fix it, to avoid exporting/importing with each upgrade.

The easiest would be if you could send the `settings3` to me over support@sqlitestudio.pl (it would let me find the problem immediately and I could also fix the settings file so it doesn't crash for you anymore), but if that's not possible, let's try approach no. 2.

Approach #2: Could you please start SQLiteStudio from `cmd` with `SQLiteStudio.exe -d --debug-file=C:\path_to_log_file_to_create.txt`
After the application crashes, please see what's in the log - is there anything that would indicate the issue? If you're not sure, you can share the log - if that's not a problem - and I can have a look at it.

There is no dedicated function to export/import configuration so far, because the app should not crash due to config (and has not been crashing because of that so far, I believe). But it may make sens to introduce such mechanism. I will think about it.

**Comment by githubuser181226** [comment]
Hello. Thank you for your quick reply, and for all the great work you've done so far.

I don't know what is happening, but after reinstalling the program anew it starts and seemingly works properly, however then it crashes when attempting to open Extension Manager.

I suspected some extension is the culprit so I started manually removing files from `ext` one by one, and found out if `percentile.dll` is gone then the crashes go away altogether.

Btw. Extension Manager has a bug where it does NOT allow you remove all the extensions, because if you empty out the list then the commit button gets disabled.

Also, I cannot send settings3 to you because it contains query logs referencing sensitive data (even database name, table names etc. I consider as such). Sorry.

On that note, IMO perhaps in future version it would be wise to make it so the program stores its configuration separately from database, especially away from actual data values (this includes any query logs, ddu history etc.). This would allow sharing settings for debug purposes without any worry of leaking any actual data (let alone sensitive data).

I attatched `debug-crash-ext-manager.log` SQLiteStudio 3.4.7 creates after crashing when opening Extension Manager with `percentile.dll` being present.
And another `debug-no-percentile-no-crash.log` after `percentile.dll` is deleted and the program works normally and closes cleanly.

Good day.

[debug-crash-ext-manager.log](https://github.com/user-attachments/files/17967333/debug-crash-ext-manager.log)
[debug-no-percentile-no-crash.log](https://github.com/user-attachments/files/17967334/debug-no-percentile-no-crash.log)

**Reply by pawelsalawa** [reply]
Oh, so it looks like a problem with the extension. "Not great, not terrible" ;) Is this extension available publicly, so I can try to crash my app too?

**Reply by pawelsalawa** [reply]
...and yes, the bug about clearing all extensions is to be fixed too.

**Reply by githubuser181226** [reply]
The problematic extension is `percentile.dll` which is supplied with the program. All I did is deleted that one file and SQLiteStudio 3.4.7 went back to working correctly without crashing.

**Reply by pawelsalawa** [reply]
Interesting. It's reproducible. I'll fix it.

**Reply by pawelsalawa** [reply]
Created #5145 for it.
